Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -310,31 +316,43 @@ public class HTTPClient {
+        /// Enables automatic body decompression. Supported algorithms are gzip and deflate.
+        public var decompression: Bool
         /// Ignore TLS unclean shutdown error, defaults to `false`.
         public var ignoreUncleanSSLShutdown: Bool
-
-        public init(tlsConfiguration: TLSConfiguration? = nil, followRedirects: Bool = false, timeout: Timeout = Timeout(), proxy: Proxy? = nil) {
-            self.init(tlsConfiguration: tlsConfiguration, followRedirects: followRedirects, timeout: timeout, proxy: proxy, ignoreUncleanSSLShutdown: false)
-        }
-
-        public init(tlsConfiguration: TLSConfiguration? = nil, followRedirects: Bool = false, timeout: Timeout = Timeout(), proxy: Proxy? = nil, ignoreUncleanSSLShutdown: Bool = false) {
+        /// Limit decompression, default is a ratio of 100.
+        public var decompressionLimit: DecompressionLimit
+
+        public init(tlsConfiguration: TLSConfiguration? = nil,
+                    followRedirects: Bool = false,
+                    timeout: Timeout = Timeout(),
+                    proxy: Proxy? = nil,
+                    ignoreUncleanSSLShutdown: Bool = false,
+                    decompression: Bool = false,
+                    decompressionLimit: DecompressionLimit = .ratio(100)) {
             self.tlsConfiguration = tlsConfiguration
             self.followRedirects = followRedirects
             self.timeout = timeout
             self.proxy = proxy
             self.ignoreUncleanSSLShutdown = ignoreUncleanSSLShutdown
+            self.decompression = decompression
+            self.decompressionLimit = decompressionLimit
         }
 
-        public init(certificateVerification: CertificateVerification, followRedirects: Bool = false, timeout: Timeout = Timeout(), proxy: Proxy? = nil) {
-            self.init(certificateVerification: certificateVerification, followRedirects: followRedirects, timeout: timeout, proxy: proxy, ignoreUncleanSSLShutdown: false)
-        }
-
-        public init(certificateVerification: CertificateVerification, followRedirects: Bool = false, timeout: Timeout = Timeout(), proxy: Proxy? = nil, ignoreUncleanSSLShutdown: Bool = false) {
+        public init(certificateVerification: CertificateVerification,
+                    followRedirects: Bool = false,
+                    timeout: Timeout = Timeout(),
+                    proxy: Proxy? = nil,
+                    ignoreUncleanSSLShutdown: Bool = false,
+                    decompression: Bool = false,
+                    decompressionLimit: DecompressionLimit = .ratio(100)) {
             self.tlsConfiguration = TLSConfiguration.forClient(certificateVerification: certificateVerification)
             self.followRedirects = followRedirects
             self.timeout = timeout
             self.proxy = proxy
             self.ignoreUncleanSSLShutdown = ignoreUncleanSSLShutdown
+            self.decompression = decompression
+            self.decompressionLimit = decompressionLimit
```

Comment:
should we squash `decompression` and `decompressionLimit` into one thing? make `decompression` an emum with one options being `none` (no decompression), then other options for various limits? or something along these lines?

Reply:
I don't think having it as enum would be a good choice, `decompressionLimit` is a configuration of `HTTPResponseDecompressor`, but having to deal with 'no decompression' in the settings of it is a bit unlcean, imo. But I agree that having two parameters is also not a good idea (what would decompression=false and limit=ratio mean?). What about making `decompression` an enum:
```
enum HTTPDecompression {
    case none
    case enabled(limit: HTTPDecompressionLimit)
}
```
wdyt?

Reply:
yes that is exactly how i imagined it

Reply:
re-factored